Orbit.h

```diff
@@ -9,45 +9,173 @@
+//     DoSomeOtherWork();
+//   }
+// }
+//
+// Parameters:
+// name: [const char*] Label to be displayed on current time slice (kMaxEventStringSize characters).
+// col: [orbit::Color] User defined color for the current time slice.
+//
 #define ORBIT_SCOPE(name) ORBIT_SCOPE_WITH_COLOR(name, orbit::Color::kAuto)
 #define ORBIT_SCOPE_WITH_COLOR(name, col) orbit_api::Scope ORBIT_VAR(name, col)
 
-// ORBIT_START/ORBIT_STOP: profile time span on a single thread.
+// ORBIT_START/ORBIT_STOP: Profile sections inside a scope.
+//
+// Overview:
+// Profile the time between ORBIT_START and ORBIT_STOP.
+//
+// Notes:
+// 1. ORBIT_START and ORBIT_STOP need to be in the same scope. For start and stop operations that
+//    need to happen in different scopes or threads, use ORBIT_ASYNC_START/ORBIT_ASYNC_STOP.
+//
+// 2. We limit the maximum number of characters of the "name" parameter to "kMaxEventStringSize".
+//    This limitation will be lifted as we roll out a new dynamic instrumentation implementation on
+//    Linux.
+//
+// Example Usage: Profile sections of a function:
+//
+// void MyVeryLongFunction() {
+//   DoSomeWork();
+//
+//   ORBIT_START("DoSomeMoreWork");
+//   DoSomeMoreWork();
+//   ORBIT_STOP();
+//
+//   ORBIT_START_WITH_COLOR("DoSomeOtherWork", orbit::Color::kLightGreen);
+//   DoSomeOtherWork();
+//   ORBIT_STOP();
+// }
+//
+// Parameters:
+// name: [const char*] Label to be displayed on current time slice (kMaxEventStringSize characters).
+// col: [orbit::Color] User defined color for the current time slice.
+//
 #define ORBIT_START(name) ORBIT_START_WITH_COLOR(name, orbit::Color::kAuto)
 #define ORBIT_START_WITH_COLOR(name, col) orbit_api::Start(name, col)
 #define ORBIT_STOP() orbit_api::Stop()
 
-// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: profile time span across threads.
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: Profile time spans across scopes or threads.
+//
+// Overview:
+// Async time spans can be started in one scope and stopped on another. They will be displayed
+// in Orbit on a track uniquely identified by the "name" parameter. Note that those time slices
+// do not represent hierarchical information.
+//
+// Note:
+// We limit the maximum number of characters of the "name" parameter to "kMaxEventStringSize". This
+// limitation will be lifted as we roll out a new dynamic instrumentation implementation on Linux.
+// It is possible however to add per-time slice additional information using the ASYNC_STRING macro.
+//
+// Example usage: Tracking "File IO" operations.
+// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);  // File IO request site.
+// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
+// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);  // File IO result site.
+//
+// Parameters:
+// name: [const char*] Name of the *track* that will display the async events in Orbit.
+// id: [uint64_t] A user-provided unique id for the time slice.
```

Comment:
I think it could help understanding here to add how the unique id is used: "This unique id is used to match start and stop calls."

IMO, that clearly emphasizes that it *has* to be unique and strange stuff will happen if it is not.